File: Problems/PascalTriangle.js
/**
 * @param {number} numRows
 * @return {number[][]}
 */
var generate = function(numRows) {
    const result = [];
    if (numRows > 0) result.push([1]);
    
    if (numRows > 1) result.push([1, 1]);
    
    for (let i = 2; i < numRows; i++) {
        result.push(result[i-1].concat(0).map((v, idx, ori) => v + ori[ori.length - idx - 1])
        );
    }
    return result;
};

File: Problems/PascalTriangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        result = [[1]]        
        for row in range(numRows-1):
            auxMatrix = [1]
            for column in range(1, row+1):
                auxMatrix.append(result[row][column-1] + result[row][column])                
            auxMatrix.append(1)
            result.append(auxMatrix)            
        return result
